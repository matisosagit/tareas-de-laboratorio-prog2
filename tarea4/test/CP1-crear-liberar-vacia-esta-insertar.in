# Test para TColaDePrioridadPerros: crear, liberar, estaVacia, esta, insertar

crearCP 11

# chequeo con cola vacía
esVaciaCP
estaEnCP 1

# creo Perro 1 (Vitalidad 10)
crearFecha 10/10/2022
crearPerro 1 Fido 3 10 Un perro amigable

# inserto el perro en la cola
insertarEnCP

# ahora la cola no está vacía
esVaciaCP

# chequeo perros en la cola
estaEnCP 1
estaEnCP 2  # No insertado aun
estaEnCP 3 # No insertado aun

# creo Perro 2 (Vitalidad 20) - menos prioritario que Perro 1
crearFecha 22/03/2023
crearPerro 2 Max 4 20 Un perro grande

# inserto en cola
insertarEnCP

# creo Perro 3 (Vitalidad 2) - el prioritario
crearFecha 01/01/2024
crearPerro 3 Rocky 2 2 Un perro pequeño y energético

# inserto en cola
insertarEnCP

# testeo perros en cola
estaEnCP 1
estaEnCP 6  # No insertado
estaEnCP 2
estaEnCP 3
estaEnCP 10  # No insertado

liberarCP

Fin
