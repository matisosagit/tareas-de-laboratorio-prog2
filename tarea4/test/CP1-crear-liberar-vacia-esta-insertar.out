1> #  Test para TColaDePrioridadPerros: crear, liberar, estaVacia, esta, insertar.
2> Se creó una cola de prioridad de perros con cantidad maxima 11
3> #  chequeo con cola vacía.
4> La cola de prioridad es vacía.
5> El perro con id 1 NO está en la cola de prioridad.
6> #  creo Perro 1 (Vitalidad 10).
7> Fecha creada en forma exitosa.
8> 9> #  inserto el perro en la cola.
10> Se insertó el perro en la cola de prioridad.
11> #  ahora la cola no está vacía.
12> La cola de prioridad NO es vacía.
13> #  chequeo perros en la cola.
14> El perro con id 1 está en la cola de prioridad.
15> El perro con id 2 NO está en la cola de prioridad.
16> El perro con id 3 NO está en la cola de prioridad.
17> #  creo Perro 2 (Vitalidad 20) - menos prioritario que Perro 1.
18> Fecha creada en forma exitosa.
19> 20> #  inserto en cola.
21> Se insertó el perro en la cola de prioridad.
22> #  creo Perro 3 (Vitalidad 2) - el prioritario.
23> Fecha creada en forma exitosa.
24> 25> #  inserto en cola.
26> Se insertó el perro en la cola de prioridad.
27> #  testeo perros en cola.
28> El perro con id 1 está en la cola de prioridad.
29> El perro con id 6 NO está en la cola de prioridad.
30> El perro con id 2 está en la cola de prioridad.
31> El perro con id 3 está en la cola de prioridad.
32> El perro con id 10 NO está en la cola de prioridad.
33> Se liberó la cola de prioridad de perros
34> Fin.
